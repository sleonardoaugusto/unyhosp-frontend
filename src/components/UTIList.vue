<template>
  <div class="col-md-12 d-flex justify-content-center flex-wrap">
    <div
      v-for="(u, i) in utis"
      :key="i"
      class="card col-md-5"
    >
      <p class="card__name">{{ u.name }}</p>
    </div>
  </div>
</template>

<script>
import utis from '@/mocks/utis'

export default {
  name: 'UTIList',
  data: () => ({
    utis: []
  }),
  created() {
    this.utis = utis.data
  }
}
</script>

<style scoped>

</style>