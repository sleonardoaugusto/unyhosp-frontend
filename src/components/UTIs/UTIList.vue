<template>
  <div class="col-md-12 d-flex justify-content-center flex-wrap">
    <router-link
      v-for="(u, i) in utis"
      :key="i"
      class="card col-md-3"
      :to="{name: 'beds', params: {utiId: u.id}}"
    >
      <div class="row mb-5">
        <p class="card__name">{{ u.name }}</p>
      </div>
    </router-link>
  </div>
</template>

<script>
import UTIService from '@/services/UTIService'

export default {
  name: 'UTIList',
  data: () => ({
    utis: []
  }),
  created() {
    const id = this.$route.params.hospitalId
    UTIService.getByHospitalId(id)
      .then((res) => {
        this.utis = res.data
      })
  }
}
</script>

<style scoped>

</style>