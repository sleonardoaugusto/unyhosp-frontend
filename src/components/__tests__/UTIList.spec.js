import { shallowMount, RouterLinkStub } from '@vue/test-utils'
import UTIList from '@/components/UTIList'
import utis from '@/components/__tests__/__mocks__/utis'


describe('HospitalList.vue', () => {
  let wrapper

  const $route = {
    params: { hospitalId: 1 }
  }

  beforeEach(() => {
    wrapper = shallowMount(UTIList, {
      stubs: {
        RouterLink: RouterLinkStub
      },
      mocks: {
        $route
      }
    })
  })

  it('component is rendering properly', () => {
    wrapper.setData({ utis: utis.data })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
