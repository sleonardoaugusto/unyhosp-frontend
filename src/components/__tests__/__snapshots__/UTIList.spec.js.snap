// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HospitalList.vue component is rendering properly 1`] = `
<div class="col-md-12 d-flex justify-content-center flex-wrap"><a class="card col-md-5">
    <p class="card__name">UTI A - São Joaquim</p>
  </a><a class="card col-md-5">
    <p class="card__name">UTI B - São Joaquim</p>
  </a><a class="card col-md-5">
    <p class="card__name">UTI C - São Joaquim</p>
  </a></div>
`;
